scripts/components/Tooltip/index.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { cloneElement, PropTypes } from 'react';
+import { render, unmountComponentAtNode, findDOMNode } from 'react-dom';
+import { addClass, removeClass } from '../../utils/DomUtils';
+import utils from './utils';
+
+const ARROW_LEFT = 'tooltip -arrowright';
+const ARROW_RIGHT = 'tooltip -arrowleft';
+const ARROW_UP = 'tooltip -arrowdown';
+const ARROW_DOWN = 'tooltip -arrowup';
+
+
+export default class ToolTip extends React.Component {
+
+  constructor(props) {
+    super(props);
+
+    this.handleMouseEnter = ::this.handleMouseEnter;
+    this.handleMouseOut = ::this.handleMouseOut;
+  }
+
+  state = {
+    show: false,
+    top: 0,
+    left: 0
+  }
+
+  static PropTypes = {
+    position: PropTypes.oneOf([
+    'top',
+    'left',
+    'right',
+    'bottom'
+    ]).isRequired,
+    text: PropTypes.string.isRequired
+  };
+
+  static defaultProps = {
+    position: 'left',
+    text: 'Input Tooltip Text here',
+  }
+
+  componentDidMount() {
+    this.mountContainer();
+    if ( this.state.show ) {
+      this.mountTooltip();
+    }
+    findDOMNode(this).addEventListener('mouseenter', this.handleMouseEnter);
+    findDOMNode(this).addEventListener('mouseout', this.handleMouseOut);
```

Comment:
As much as possible, we'd like to avoid using the dom node itself. In this case, you can clone the trigger, and add the `mouseenter` and `mouseout` listeners.

For example:
```es6
return clloneElement(this.props.children, {
  onMouseEnter: ::this.handleMouseEnter
  onMouseOut: ::this.handleMouseOut
});
```
<sup> Taken from [suleras/OverlayTrigger#L78](https://github.com/sourcescript/suleras/blob/67cd6a7b7b36f3f3f7d52bb59d11022e875163d0/scripts/components/Tooltip/OverlayTrigger.js#L78) </sup>